shortened menu keypress delays, date fmt D/M/Y changed to D.M.Y

clixt opened the pull request:
- Date format MDY is displayed as M/D/Y, format DMY is displayed as D.M.Y. This way it's easier to visually recognize the chosen format.
- KeyPress times for EVENT_ENTER_MENUS, EVENT_EXIT_MENUS and EVENT_PARENT_MENU shortened to 200ms, since there is no real need to wait longer than that.
- removed some brackets

ClockSelect commented:
I'm using your bin.

D/M/Y is the graphism civilized people are used to; why would you change it? What's more, it's the default format, so it's fine. The date no longer looks like a date; it's disturbing.

200ms is indistinguishable from instant. A sensible delay is comfortable and needed, imo.

Why would you remove just two brackets? There's thousands of them.

clixt commented:
> "I'm using your bin."

Nice, you're really quick!  
I removed the binfile from my repo, now it's easier to merge your origin. And so people don't accidentally install my bin instead of yours ;)

The rest gets little more nerdy (sorry).

> "D/M/Y is the graphism civilized people are used to"

Out of 138 countries, only 15 use the DMY format with the "/" component separator (graphism), while 34 countries use the "." separator (including Austria, Germany, Finland, Poland, Russia and Switzerland). Unfortunately, Wikipedia makes no assumptions about how "civilized" these countries are, nor how many vapers they might have.
The rest uses either a different component order, or there is no conclusive information about the used separator.
(source: see https://en.wikipedia.org/wiki/Date_format_by_country)

My opinion:
The only "civilized" format is YYYY-MM-DD as defined by ISO8601, since others are not standardized.

> "why would you change it?"

It is safe to conclude:
For the middle-endian MDY component order, the "/" separator is used pretty much always.
The "." separator is used by the majority of countries which use little-endian (DMY) order.
Therefore, "D.M.Y" and "M/D/Y" option seem to be the most widely accepted design choice.
Also, as I wrote before, this way it's much easier to immediately visually recognize the chosen format: "3.8.2016" always means "third august". If you use the same "/" separator for both "MDY" and "DMY", then "3/8/2016" looks like "third august" as well as "march eight", which could be confusing or disturbing to people in 34 countries.

> "What's more, it's the default format, so it's fine"

Yes, it's the default, but it sucks. Lots of defaults suck, that is why people put a lot of effort into making a custom suck-less firmware.

> "The date no longer looks like a date; it's disturbing."

Wikipedia says: it does not look like a date in 15 countries, it looks very much like a date in 34 countries. See link above.

> "200ms is indistinguishable from instant. A sensible delay is comfortable and needed, imo."

Yes, 200ms feels instant, snappy, and responsive, that is how I like my interfaces. I understand how most people (especially Winblows users) might be used to much higher "sensible delays" and feel comfortable with much slower response times. 

However, I don't see why a longer delay is "needed", please explain. Does this collide with some other functionality? I experience no issues with 200ms response times on my MOD so far.

> "Why would you remove just two brackets? There's thousands of them."

Totally true. In addition to thousands useless brackets there are many badly-nested if structures,
cryptically named global variables, one over-500-lines-long select case, and other fun stuff. The codebase seems to be a little mess, probably thanks to it's humble disassembly origins. 

I just removed these brackets at random, not that it's important. Just playing with the code. No real reason.

ClockSelect commented:
Let's give it a try!
